### PR TITLE
feat: Pre-release patches toggle

### PR DESCRIPTION
**Pull request by kitadai31**
Add a toggle for pre-release patches to the alternative source settings.
Closes: #821

This PR also fixes an problem where pre-release Integrations are being used when using alternate sources.
At this time, Patches are referring `GitHubAPI#getLatestPatchesRelease()` that doesn't include pre-releases, but Integrations are referring `GitHubAPI#getLatestRelease()` that includes pre-releases.
This problem actually caused patched apps to crash recently in crimera/piko patches (an alternative source) when they released a pre-releases.
(Actually, this is the main motivation)

`getLatestReleaseFile()` in `github_api.dart` was removed because it's unused

### CODE REVIEW

**Comment by Ushie** [comment]
Why limit it to alternative sources?

You can pass ?dev=true to achieve the same with the ReVanced API

**Comment by kitadai31** [comment]
I already tried ?dev=true at ReVanced API, but it didn't effect.
https://api.revanced.app/v2/revanced-patches/releases/latest?dev=true
So, I thought it wasn't implemented yet at first.
That's why it is limited to GitHub API only.  I didn't know the cache issue.
If the cache issue will be fixed, I think I should wait for it.

> Are changelogs/updates considered as well?

Yes. These lines change the patches update dialog according to the settings.
https://github.com/ReVanced/revanced-manager/blob/e88e997436d8692023d49480089fe7d71538492b/lib/ui/views/home/home_viewmodel.dart#L482-L490

**Comment by brian6932** [comment]
One thing I noticed after building this branch (rebased) was that RevancedManager, gets stuck on `Loading...`, patches work fine though. Is that just a quirk of debug builds, or an issue with the PR?

**Comment by nivranaitsirhc** [comment]
> One thing I noticed after building this branch (rebased) was that checking for `Revanced Manager` updates gets stuck on `Loading...`, patches work fine though. Is that just a quirk of debug builds, or an issue with the PR?

Can you share the apk here please

**Comment by brian6932** [comment]
@nivranaitsirhc sure https://i.alienpls.org/ReVanced.apk

**Comment by nivranaitsirhc** [comment]
> @nivranaitsirhc sure https://i.alienpls.org/ReVanced.apk

It works like a charm! Thanks!

**Comment by kitadai31** [comment]
Waiting for the parameter issue in ReVanced API to be fixed

I have some questions

1. I added the ReVanced API v2 related logic in revanced_api.dart, but should I split the file into two files like `revanced_api_v2.dart` ?
2. I added a word "(dev)" in settings description because ReVanced calls pre-release versions “dev”. (40b1589)
Is this no problem?

File: lib/services/github_api.dart [hunk 1]
```diff
@@ -37,6 +37,34 @@ class GithubAPI {
...
+      }
+      return latestRelease;
+    } on Exception catch (e) {
+      if (kDebugMode) {
+        print(e);
+      }
+      return null;
+    }
+  }
+
```
Comment (Ushie) [inline lib/services/github_api.dart:67]: Maybe instead of making a new function, add a parameter to the existing one instead

**Review by Ushie — CHANGES_REQUESTED** [review]
@oSumAtrIX Does v3 suffer from the same issues regarding the ?dev parameter, supporting the API is a prerequisite for this PR

**Comment by oSumAtrIX** [comment]
A ?dev query parameter does not exist and is not planned to be added. The TLDR is that the dev branch is not a controlled release channel like a "beta" channel would be.

**Comment by Ushie** [comment]
The dev branch has nothing to do with this, the flag can be renamed to ?prereleases instead of ?dev because it simply enables returning prereleases

**Comment by oSumAtrIX** [comment]
The name does not change that pre releases are not a controlled production release channel. Any further discussion should happen in the API repo and not in this PR

**Comment by kitadai31** [comment]
Yeah, I've heard that ReVanced API v2 was deprecated.
And v3 doesn't have toggles for pre-releases.

So, I think it's enough to revert this PR to the beginning. (Pre-release requires alternative sources to be enabled, https://github.com/ReVanced/revanced-manager/pull/1785#issue-2205094068)

**Pull request closed without merging**